### PR TITLE
Add button helper + Form::ButtonComponent

### PR DESCRIPTION
**Pull request by Spone**
Related to #4

### CODE REVIEW

File: app/components/view_component/form/button_component.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    class ButtonComponent < BaseComponent
+      attr_reader :value
+
+      def initialize(form, object_name, value, options = {}, &block)
+        @value = value
+
+        if block_given?
+          value = capture { yield(value) }
+        end
+
```
Comment (nicolas-brousse) [inline app/components/view_component/form/button_component.rb:14]:
```suggestion
```

File: app/components/view_component/form/button_component.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        super(form, object_name, options)
+      end
+
+      def call
+        button_tag(value, options)
```
Comment (nicolas-brousse) [inline app/components/view_component/form/button_component.rb:19]:
```suggestion
        button_tag(content || value, options)
```

File: lib/view_component/form/builder.rb [hunk 3]
```diff
@@ -55,8 +55,14 @@ def submit(value = nil, options = {})
...
-      # def button(value = nil, options = {}, &block)
-      # end
+      def button(value = nil, options = {}, &block)
+        if value.is_a?(Hash)
+          options = value
+          value = nil
+        end
+        value ||= submit_default_value
+        render_component(:button, self, @object_name, value, options, &block)
+      end
```
Comment (nicolas-brousse) [inline lib/view_component/form/builder.rb:65]: Maybe we could remove `@object_name` like for submit

**Review by nicolas-brousse — CHANGES_REQUESTED** [review]
I think we should add a test with a block for value.

File: lib/view_component/form/builder.rb (hunk 3 above)
Comment (Spone) [reply lib/view_component/form/builder.rb]: Done in 587b550

**Comment by Spone** [comment]
> I think we should add a test with a block for value.

Done in 30688d1 but the test is failing. Any idea why?

**Review by nicolas-brousse — APPROVED** [review]

**Pull request merged**